src/ast_util.py

```diff
@@ -166,6 +166,126 @@ def generic_visit(self, node: ca.Node) -> None:
     return Indices(starts, ends)
 
 
+def get_noncolliding_fn_name(ast: ca.FileAST, name: str) -> str:
+    used_names: Set[str] = set()
+    for item in ast.ext:
+        if isinstance(item, ca.Decl) and item.name:
+            used_names.add(item.name)
+        if isinstance(item, ca.FuncDef) and item.decl.name:
+            used_names.add(item.decl.name)
+    return get_noncolliding_name(used_names, name)
+
+
+def get_noncolliding_struct_name(typemap: TypeMap, name: str) -> str:
```

Comment:
Instead of two different functions I'd suggest a single one that works for both struct and function names (and variables, ...). Function names and typedefs share the same namespace anyway.

Reply:
I split the functions by namespace